cdap-tms/src/main/java/co/cask/cdap/messaging/client/ClientMessagingService.java

```diff
@@ -342,8 +342,16 @@ private ByteBuffer encodeRollbackDetail(RollbackDetail rollbackDetail) throws IO
    * Verifies the content-type in the header matches with the required type.
    */
   private void verifyContentType(Map<String, ? extends Collection<String>> headers, String requiredContentType) {
-    Collection<String> values = headers.get(HttpHeaders.CONTENT_TYPE);
-    if (values == null || !requiredContentType.equals(Iterables.getFirst(values, null))) {
+    // Netty 4.1 has all headers set with lower case name
+    // However, the cdap-http library doesn't handle it well, hence we iterate all headers instead of lookup in here
+    boolean passed = false;
+    for (Map.Entry<String, ? extends Collection<String>> entry : headers.entrySet()) {
+      if (HttpHeaders.CONTENT_TYPE.equalsIgnoreCase(entry.getKey())) {
+        passed = requiredContentType.equalsIgnoreCase(Iterables.getFirst(entry.getValue(), null));
+        break;
```

Comment:
If we break here, can't we just move the below condition and its `throw new IllegalArgumentException` here?
Seems simpler to read that way.

Reply:
It break when it found the `content-type` header, regardless of case. But it may or may not pass depending on the value of the header. If I move the `if (!pass) throw ...` here, then after the for-loop, I need to have another `throw ...` statement. Seems duplicate to me.

Reply:
ah ok. I didn't realize that the exception would be thrown, even if there's no content type header.